### ISSUE TITLE
Redo syntax for case 

### ISSUE THREAD

**Issue by byorgey** [disco-lang/disco#43]
Reasons for getting rid of current case syntax:

1. We want to use `{` for sets
2. It's not possible to write a multi-branch case on a single line
3. The layout parsing is hacky and broken.

Proposed new syntax:

```
{? t guards
, t guards
, ...
, t guards 
?}
```
but of course the newlines are not required.

As a separate issue, once this is done we should rip out the layout parsing and just deal with parsing documentation etc. manually.
